lib/ansible/template/safe_eval.py

```diff
@@ -139,11 +141,15 @@ def generic_visit(self, node, inside_call=False):
     try:
         parsed_tree = ast.parse(expr, mode='eval')
         cnv.visit(parsed_tree)
-        compiled = compile(parsed_tree, expr, 'eval')
+        compiled = compile(parsed_tree, to_native(expr), 'eval')
```

Comment:
Do we want errors with this?
```suggestion
        compiled = compile(parsed_tree, to_native(expr, errors='surrogate_or_strict'), 'eval')
```

Reply:
I'm not sure if that makes sense. I want to look at this some more tomorrow. Today was hectic. 🤯

Reply:
I don't think we need `errors='surrogate_or_strict'` for that. It's `filename` argument for the `compile` function [0]:

> The filename argument should give the file from which the code was read; pass some recognizable value if it wasn’t read from a file ('<string>' is commonly used).

So the value is information only and we are fine with omitting "bad bytes" from the value.

[0] https://docs.python.org/3/library/functions.html#compile

Reply:
Ok, that makes sense. Thanks for the great explanation.